CDCgov/prime-simplereport
# Johanna/5113 create new function app fhir test events

# DEVOPS PULL REQUEST

## Related Issue

Implements #5113 and #5114

## Changes Proposed

The objective of this PR is to create a new function `FHIRTestEventReporter` within the `report_stream_batched_publisher` function app that will take care of reading and processing the FHIR multiplex test events pushed in the `fhir-data-publishing` queue. The code changes implemented take care of below items:
- Creates a new function within the function app `report_stream_batched_publisher`
- Modularizes the logic used by the function `QueueBatchedTestEventPublisher` so it can be reused by the new function
- Updates current unit testing to reflect the new logic split
- Implements the logic that takes care of reading the FHIR test events from `fhir-data-publishing` queue and process them into object type `ProcessedTestEvents`
- Implements new unit testing for the new logic 
- Add terraform changes to create the function and sets up env variables to refer to the queues `fhir-data-publishing` and `fhir-data-publishing-exceptions` used by the new function
- Updates the left-hook so the eslint checks, formatting and fixes trigger as pre-commit step

## Additional Information
- The logic to actually send the events to ReportStream will be done with ticket 5115 but you can see a rough implementation as part of this ticket but the code is commented out.
 
- You will see some code changes that are just formatting updates because of the left-hook triggering for updates in this folder moving forward.

- As part of this PR it has been decided that the reporting of FHIR test events will use its own queues to read and report failed parse test events but it will reuse the existing queue to log ReportStream response errors. 

## Testing
_This PR needs to be verified in a test environment (I am testing in dev4)_
- Verify that the existing logic to report test events works as expected
- Verify that the events published to `fhir-data-publishing` are correctly read and processed. These events will not be sent to the ReportStream yet so to validate this just check that the context logs are printed as expected in azure.
